crates/cranelift/src/func_environ.rs

```diff
@@ -1573,9 +1589,12 @@ impl<'module_environment> cranelift_wasm::FuncEnvironment for FuncEnvironment<'m
         index: ir::Value,
     ) -> WasmResult<ir::Value> {
         let plan = &self.module.table_plans[table_index];
+        let table = plan.table;
+        let index_type = table.idx_type;
+        let index = self.cast_index_to_i64(&mut builder.cursor(), index, index_type);
```

Comment:
I think that this may be roughly where the verifier error is coming from. We only want to cast indices to `i64` when it's going out to a libcall, but otherwise the index will stay in its native type.